Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -1376,6 +1377,60 @@ bool RenderStyle::changeRequiresRecompositeLayer(const RenderStyle& other, Optio
     return false;
 }
 
+bool RenderStyle::scrollAnchoringSuppressionStyleDidChange(const RenderStyle* other) const
+{
+    // https://drafts.csswg.org/css-scroll-anchoring/#suppression-triggers
+    // Determine if there are any style changes that should result in an scroll anchoring suppression
+    if (!other)
+        return false;
+
+    if (m_nonInheritedData->boxData.ptr() != other->m_nonInheritedData->boxData.ptr()) {
+        if (m_nonInheritedData->boxData->width() != other->m_nonInheritedData->boxData->width()
+            || m_nonInheritedData->boxData->minWidth() != other->m_nonInheritedData->boxData->minWidth()
+            || m_nonInheritedData->boxData->maxWidth() != other->m_nonInheritedData->boxData->maxWidth()
+            || m_nonInheritedData->boxData->height() != other->m_nonInheritedData->boxData->height()
+            || m_nonInheritedData->boxData->minHeight() != other->m_nonInheritedData->boxData->minHeight()
+            || m_nonInheritedData->boxData->maxHeight() != other->m_nonInheritedData->boxData->maxHeight())
+            return true;
+    }
+
+    if (overflowAnchor() != other->overflowAnchor() && overflowAnchor() == OverflowAnchor::None)
+        return true;
+
+    if (position() != other->position())
+        return true;
+
+    if (m_nonInheritedData->surroundData != other->m_nonInheritedData->surroundData) {
```

Comment:
Should this check `m_nonInheritedData->surroundData.ptr()`?

Reply:
Fixed.